app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee

```diff
@@ -144,18 +144,14 @@ ShortFormApplicationController = (
 
   $scope.handleFormSuccess = ->
     options = ShortFormNavigationService.submitOptionsForCurrentPage()
-    if options.callback
-      options.callback.forEach (callback) ->
-        $scope[callback](options.params) if $scope[callback]
+    if options.callbacks
+      options.callbacks.forEach (callback) ->
+        callback()
+    if options.scopedCallbacks
+      options.scopedCallbacks.forEach (scopedCallback) ->
+        $scope[scopedCallback.func](scopedCallback.param) if $scope[scopedCallback.func]
     if options.path
-      $scope.goToAndTrackFormSuccess(options.path)
-
-  $scope.goToAndTrackFormSuccess = (path, params) ->
-    AnalyticsService.trackFormSuccess('Application')
```

Comment:
Seems like the analytics service here is a bit of a side-effect of this function, not actually part of the goal. Would it be clearer if we just called it goToPage? I wasn't sure what "TrackFormSuccess" was just looking at where we're calling the function.